core/src/main/java/org/elasticsearch/action/search/TransportMultiSearchAction.java

```diff
@@ -154,20 +162,32 @@ private void handleResponse(final int responseSlot, final MultiSearchResponse.It
                 } else {
                     if (thread == Thread.currentThread()) {
                         // we are on the same thread, we need to fork to another thread to avoid recursive stack overflow on a single thread
-                        threadPool.generic().execute(() -> executeSearch(requests, responses, responseCounter, listener));
+                        threadPool.generic().execute(() -> executeSearch(requests, responses, responseCounter, startTimeInNanos, listener));
                     } else {
                         // we are on a different thread (we went asynchronous), it's safe to recurse
-                        executeSearch(requests, responses, responseCounter, listener);
+                        executeSearch(requests, responses, responseCounter, startTimeInNanos, listener);
                     }
                 }
             }
 
             private void finish() {
-                listener.onResponse(new MultiSearchResponse(responses.toArray(new MultiSearchResponse.Item[responses.length()])));
+                listener.onResponse(new MultiSearchResponse(getExecTimeInMillis(startTimeInNanos), responses.toArray(new MultiSearchResponse.Item[responses.length()])));
             }
+            
+            /**
+             * Builds how long it took to execute the msearch.
+             */
+            private long getExecTimeInMillis(long startTimeNanos) {
+                return TimeUnit.NANOSECONDS.toMillis(getRelativeTime() - startTimeNanos);
+            }
+
         });
     }
 
+    private long getRelativeTime() {
```

Comment:
In `TransportBulkAction`, the corresponding method is named `relativeTime`; let's do the same here.